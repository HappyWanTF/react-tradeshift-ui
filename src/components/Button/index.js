import PropTypes from 'prop-types';
import React from 'react';

const Button = ({ children, busy, type, ...props }) => {
	const buttonProps = { ...props, 'data-ts.busy': busy };

	// @See https://github.com/yannickcr/eslint-plugin-react/issues/1555
	/* eslint-disable react/button-has-type */
	return (
		<button data-ts="Button" type={type} {...buttonProps}>
			{children}
		</button>
	);
	/* eslint-enable react/button-has-type */
};

Button.propTypes = {
	children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
	busy: PropTypes.bool,
	type: PropTypes.string
};

Button.defaultProps = {
	children: null,
	busy: false,
	type: 'button'
};

export default Button;
