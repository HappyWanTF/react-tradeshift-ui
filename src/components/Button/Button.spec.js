import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Button from '.';

configure({ adapter: new Adapter() });

describe('Button', () => {
	it('should render correctly', () => {
		const wrapper = shallow(<Button>Default button</Button>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should render correctly having className', () => {
		const wrapper = shallow(<Button className="ts-primary">Default button</Button>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should set button state to true correctly', () => {
		const button = shallow(<Button busy>busy button</Button>);

		expect(button.prop('data-ts.busy')).toBe(true);
	});
});
