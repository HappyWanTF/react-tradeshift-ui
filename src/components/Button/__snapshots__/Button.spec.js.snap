// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  data-ts="Button"
  data-ts.busy={false}
  type="button"
>
  Default button
</button>
`;

exports[`Button should render correctly having className 1`] = `
<button
  className="ts-primary"
  data-ts="Button"
  data-ts.busy={false}
  type="button"
>
  Default button
</button>
`;
